compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/AMDAIEInsertCores.cpp

```diff
@@ -32,15 +32,14 @@ namespace {
 
 /// Utility which returns 'true' is the operation needs to be inserted with an
 /// `amdaie.core` op.
-/// Some ops are surrrounded by scf.for loop nests. Place the entire
+/// Some ops are surrounded by scf.for loop nests. Place the entire
 /// loop nest inside the amdaie.core op here. Currently look for a
 /// subset of ops which we know should be in the core.
 /// TODO(newling) improve this design.
 static bool isCoreComputeOp(Operation *op) {
   return isa<linalg::LinalgOp, vector::ContractionOp,
-             memref::ExtractStridedMetadataOp, func::CallOp, arith::ExtFOp,
-             arith::TruncFOp, arith::TruncIOp, vector::TransferReadOp,
-             vector::TransferWriteOp>(op);
+             memref::ExtractStridedMetadataOp, func::CallOp,
```

Comment:
Not necessarily, see matmul + truncf:

```
// input ${M}x${K}x${TYPE1}
// input ${K}x${N}x${TYPE1}

func.func @matmul_truncf(%arg0: tensor<${M}x${K}x${TYPE1}>, %arg1: tensor<${K}x${N}x${TYPE1}>) -> tensor<${M}x${N}x${TYPE1}>
{
  %cst = arith.constant ${ZERO} : ${TYPE2}
  %0 = tensor.empty() : tensor<${M}x${N}x${TYPE2}>
  %1 = linalg.fill ins(%cst : ${TYPE2}) outs(%0 : tensor<${M}x${N}x${TYPE2}>) -> tensor<${M}x${N}x${TYPE2}>
  %2 = linalg.matmul ins(%arg0, %arg1 : tensor<${M}x${K}x${TYPE1}>, tensor<${K}x${N}x${TYPE1}>)
    outs(%1: tensor<${M}x${N}x${TYPE2}>) -> tensor<${M}x${N}x${TYPE2}>
  %3 = arith.truncf %2 : tensor<${M}x${N}x${TYPE2}> to tensor<${M}x${N}x${TYPE1}>
  return %3: tensor<${M}x${N}x${TYPE1}>
}
```
Not sure how this PR is passing though...

Reply:
I see, thanks, that's great! Then we can assume all computational ops will be linalg ops, which makes these checks a lot less fragile and improves maintainability.

Reply:
Yes!